File: dohyeon/Programmers/Level_2/Lv2_스킬트리.py
import re
def solution(skill, skill_trees):
    answer = 0
    re_skill_tree = []
    pattern = r"[^{}]".format(skill)
    for i in skill_trees :
        result = re.sub(pattern, "", i)
        re_skill_tree.append(result)
    for skills in re_skill_tree:
        idx = 0
        possible = True
        for s in skills:
            if s in skill:
                if s != skill[idx]:
                    possible = False
                    break
                idx += 1
        if possible:
            answer += 1
    return answer

File: seongwon/Programmers/Level_2/Lv2_스킬트리.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int solution(string skill, vector<string> skill_trees) {
    int answer = skill_trees.size();
    for(string s : skill_trees){
        int idx = 0;
        for(char c : s){
            auto i = find(skill.begin(), skill.end(), c);
            if(i != skill.end()){
                if(skill[idx] != *i){
                    answer--;
                    break;
                }
                else {
                    idx++;
                }
            }
        }
    }
    return answer;
}
